tensorbay/client/job.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python3
+#
+# Copyright 2021 Graviti. Licensed under MIT License.
+#
+
+"""Basic structures of asynchronous jobs."""
+
+from typing import Any, Callable, Dict, Optional, Tuple, Type, TypeVar
+
+from tensorbay.utility import AttrsMixin, ReprMixin, ReprType, attr, camel, common_loads
+
+
+class Job(AttrsMixin, ReprMixin):  # pylint: disable=too-many-instance-attributes
+    """This class defines :class:`Job`.
+
+    Arguments:
+        update_info: The function to get the latest job information.
+        title: Title of the Job.
+        job_id: ID of the Job.
+        job_type: Type of the Job.
+        arguments: Arguments of the Job.
+        created_at: The time when the Job is created.
+        started_at: The time when the Job is started.
+        finished_at: The time when the Job is finished.
+        status: The status of the Job.
+        error_message: The error message of the Job.
+        result: The result of the Job.
+        description: The description of the Job.
+
+    """
+
+    _T = TypeVar("_T", bound="Job")
+
+    _repr_type = ReprType.INSTANCE
+    _repr_maxlevel = 2
+
+    _repr_attrs: Tuple[str, ...] = (
+        "title",
+        "arguments",
+        "created_at",
+        "started_at",
+        "finished_at",
+        "status",
+        "error_message",
+    )
+
+    title: str = attr()
+    job_id: str = attr(key=camel)
+    job_type: str = attr(key=camel)
+    arguments: Dict[str, Any] = attr()
+    created_at: int = attr(key=camel)
+    started_at: int = attr(key=camel)
+    finished_at: Optional[int] = attr(key=camel)
+    status: str = attr()
+    error_message: str = attr(key=camel)
+    _result: Optional[Dict[str, Any]] = attr(key="result")
+    description: Optional[str] = attr(default="")
+
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        update_info: Callable[[str], Dict[str, Any]],
```

Comment:
`update_info` should be private, also we use the word ends with `er` for callable object.
that means: `update_info` -> `info_updater`.
I think `job_updater` may be better.

Reply:
And this variable should also be put in implementation rather than design.